mvcc/kvstore.go

```diff
@@ -160,6 +166,56 @@ func (s *store) Hash() (hash uint32, revision int64, err error) {
 	return h, s.currentRev, err
 }
 
+func (s *store) HashByRev(rev int64) (hash uint32, currentRev int64, compactRev int64, err error) {
+	readTx := s.b.ReadTx()
+	for {
```

Comment:
Two possible solutions:
* open the read txn when there's no compaction scheduled (pricey since it will have to forcecommit, but simpler)
* hash all revisions >=compactRev, then scan the key index to get keys at the current rev with modrev < compactRev.

There should be a test to trigger this inflight behavior in the mvcc package. It doesn't look like that integration test will hit the problem very often...